### PR TITLE
update flang to 11.0.1

### PR DESCRIPTION
**Pull request by xoviat**
Also, use stock clangdev, llvmdev, and remove flang-meta package.

### CODE REVIEW

**Comment by xoviat** [comment]
@conda-forge-admin Please rerender.

**Comment by xoviat** [comment]
@isuruf a little help? https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=269656&view=logs&j=a70f640f-cc53-5cd3-6cdc-236a1aa90802&t=f5d15007-a01c-5ad8-c9ce-4d519d3b275f&l=107083

The current scheme creates a new environment, which seems to break the ninja paths.

**Comment by xoviat** [comment]
@isuruf cmake appears to install to "build env" rather than "h env". I tried to fix it, but failed.

**Comment by isuruf** [comment]
Can you revert to 6c280b5862f36a6b0caa034e0a08bf823507baee ? That run failed because llvmdev 11 is not patched to produce a `flang.exe`.

**Comment by xoviat** [comment]
@isuruf That's not correct. I reconfigured flang so that it doesn't require a patched clang or llvm. See https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=269696&view=logs&j=a70f640f-cc53-5cd3-6cdc-236a1aa90802&t=f5d15007-a01c-5ad8-c9ce-4d519d3b275f&l=107152, (scroll down to 107180). 

That build failed because it copied the files to `build_env`

```
2021-01-26T19:54:26.4517925Z -- Install configuration: "Release"
2021-01-26T19:54:26.4529322Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang
2021-01-26T19:54:26.4537354Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/ieee_arithmetic.mod
2021-01-26T19:54:26.4576860Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/ieee_arithmetic_la.mod
2021-01-26T19:54:26.4614130Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/ieee_exceptions.mod
2021-01-26T19:54:26.4640232Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/ieee_exceptions_la.mod
2021-01-26T19:54:26.4666674Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/ieee_features.mod
2021-01-26T19:54:26.4677048Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/iso_c_binding.mod
2021-01-26T19:54:26.4693191Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/iso_fortran_env.mod
2021-01-26T19:54:26.4703503Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/omp_lib.mod
2021-01-26T19:54:26.4731586Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/omp_lib_kinds.mod
2021-01-26T19:54:26.4739211Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/__norm2.mod
2021-01-26T19:54:26.4757201Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/__norm2_i8.mod
2021-01-26T19:54:26.4771664Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/include/flang/omp_lib.h
2021-01-26T19:54:26.4800056Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/flangADT.lib
2021-01-26T19:54:26.4827723Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/flangArgParser.lib
2021-01-26T19:54:26.4880884Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/libpgmath.lib
2021-01-26T19:54:26.7155734Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/pgmath.lib
2021-01-26T19:54:27.0009780Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/ompstub.lib
2021-01-26T19:54:27.0030530Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/libompstub.lib
2021-01-26T19:54:27.0058330Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/bin/libompstub.dll
2021-01-26T19:54:27.0087925Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/flangrti.lib
2021-01-26T19:54:27.0142021Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/libflangrti.lib
2021-01-26T19:54:27.0171739Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/bin/libflangrti.dll
2021-01-26T19:54:27.0215226Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/flang.lib
2021-01-26T19:54:27.3065040Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/libflang.lib
2021-01-26T19:54:27.3438161Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/bin/libflang.dll
2021-01-26T19:54:27.7056156Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/lib/flangmain.lib
2021-01-26T19:54:27.7099783Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/bin/flang.exe
2021-01-26T19:54:27.7507193Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/bin/flang1.exe
2021-01-26T19:54:27.9166422Z -- Installing: D:/bld/flang-split_1611689514148/_build_env/Library/bin/flang2.exe
```

we see that two files are copied here:

```
2021-01-26T19:54:28.3521007Z Resource usage statistics from bundling flang:
2021-01-26T19:54:28.3521541Z    Process count: 5
2021-01-26T19:54:28.3521912Z    CPU time: Sys=0:00:00.8, User=0:00:00.5
2021-01-26T19:54:28.3522334Z    Memory: 88.3M
2021-01-26T19:54:28.3522646Z    Disk usage: 499.3M
2021-01-26T19:54:28.3523031Z    Time elapsed: 0:00:07.2
2021-01-26T19:54:28.3523301Z 
2021-01-26T19:54:28.3523571Z 
2021-01-26T19:54:28.3736325Z number of files: 2
2021-01-26T19:54:28.3765465Z Fixing permissions
```

File: recipe/install_flang.bat [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" amd64
-nmake install
+
+cmake --build . --target install
 if errorlevel 1 exit 1
 
-rm %LIBRARY_BIN%\flang.dll
-rm %LIBRARY_BIN%\flangrti.dll
-rm %LIBRARY_BIN%\ompstub.dll
+rm %LIBRARY_BIN%\libflang.dll
```
Comment (isuruf) [inline recipe/install_flang.bat:7]: Why this change? Is `libflang.lib` the static library and `flang.lib` the import library for the shared library?

Comment (xoviat) [reply recipe/install_flang.bat:7]: libflang is the dynamic library. flang is the dynamic library. Of course it can be changed, but that's how it was when I pulled out my old repository.

**Comment by xoviat** [comment]
Note: it shouldn't be too difficult to get linux working with this as well, but I don't have a linux box and I'm currently too lazy to install wsl.

File: recipe/install_flang.bat (hunk 1 above)
Comment (isuruf) [reply recipe/install_flang.bat:7]: > libflang is the dynamic library. flang is the dynamic library. 

I don't understand

**Comment by xoviat** [comment]
@isuruf before you merge this, I have some minor fixes, so I'd like to update the commit. Let me know if you're okay with this first.

**Comment by xoviat** [comment]
image
image

File: recipe/install_flang.bat (hunk 1 above)
Comment (xoviat) [reply recipe/install_flang.bat:7]: see images below

Comment (xoviat) [reply recipe/install_flang.bat:7]: oh, sorry typo!

Comment (isuruf) [reply recipe/install_flang.bat:7]: Previously `flang.lib` was the import library for the dynamic library, `flang.dll`, the dynamic library and `libflang.lib` was the static library. Why change this?

**Comment by xoviat** [comment]
anything else?

**Comment by xoviat** [comment]
Wait libflang needs pgmath as well

**Comment by xoviat** [comment]
@isuruf This looks good.

**Comment by xoviat** [comment]
@isuruf I went ahead an confirmed that this works on OpenBLAS with enhanced testing (xianyi/OpenBLAS#3088).

**Comment by isuruf** [comment]
Does `flang.dll` that this produce compatible with the older `flang.dll`?

**Comment by xoviat** [comment]
I can say with complete certainty that the flang.dll that this produces is ABI-compatible with the DLL from the official flang project (if one were to be generated). The official flang project extracted some of the math functionality into a new "pgmath.dll" library. This is not a change that I made; I simply have as few changes as possible with the official flang project, while fixing any major bugs.

**Comment by xoviat** [comment]
see https://github.com/flang-compiler/flang/commit/7a601b06f965526de8a70565416dc3b8e4866e94#diff-9ce46d50a39e8b38c4ffebdeebfc08c3f1c7b8c3440cb7efada77fbb1b484829

**Comment by xoviat** [comment]
I think it would be nice to have a new flang though, because right now the clang version is stuck at 5, and flang is the only fortran compiler in conda-forge.

**Comment by xoviat** [comment]
@isuruf Is there anything else?

**Comment by isuruf** [comment]
Thanks
